### PR TITLE
Update fee card designs to show savings and MM fee

### PR DESCRIPTION
**Pull request by danjm**
Updates the fee card on the view quote screen to match the latest designs.

Edit: screenshots can be seen in this comment https://github.com/MetaMask/metamask-extension/pull/9629#issuecomment-723248306

This is blocked by https://github.com/MetaMask/metamask-extension/pull/9611

The UI implemented in this PR can stand on its own, but to fully match latest designs, it needs to be combined with https://github.com/MetaMask/metamask-extension/pull/9612

One final missing piece for latest designs is the info tooltip next to savings, which will be implemented in a forthcoming PR

### CODE REVIEW

**Comment by darkwing** [comment]
This looks awesome!  Quick question -- do we want to synchronize these two percentages?  1% is a bit more than 0.875:

PercentageDifference

File: ui/app/pages/swaps/view-quote/view-quote.js
Comment (darkwing) [inline ui/app/pages/swaps/view-quote/view-quote.js:421]: nit: this needs to be removed.

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (brad-decker) [inline ui/app/pages/swaps/fee-card/fee-card.js:164]: missing fee variable which will resolve @darkwing 's comment about the fees being different. The $1 is the replacement placeholder in our i18n solution. Do we have a checker for these types of issues?

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (brad-decker) [inline ui/app/pages/swaps/fee-card/fee-card.js:58]: request: should probably be a `p` tag instead of a `div`

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (brad-decker) [inline ui/app/pages/swaps/fee-card/fee-card.js:65]: request: should probably be a span or paragraph

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (brad-decker) [inline ui/app/pages/swaps/fee-card/fee-card.js:161]: request: use the wrapperClassName on `InfoTooltip` for this one.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (danjm) [reply ui/app/pages/swaps/fee-card/fee-card.js:164]: > Do we have a checker for these types of issues?

You mean as part of our build process? No we don't, but there is an error that gets rendered to the console in these cases.

**Comment by danjm** [comment]
@brad-decker This has been rebased and your comments have been addressed

**Review by brad-decker — DISMISSED** [review]
Changes LGTM. I would point out that there is a larger problem in our codebase as a whole about the proper use of HTML tags to match the intent. It wasn't introduced in this PR and progress has been made on some lines of change requests during the review. However, a concerted effort to fix these sort of issues is going to be needed in the future (not specific to swaps or recent work).

**Comment by danjm** [comment]
I want to block merging of this on https://github.com/MetaMask/metamask-extension/pull/9662, so that we can properly QA the displayed savings amount. Once that is merged, this will be rebased. I don't think further code changes will be needed.

**Comment by brad-decker** [comment]
as I was reviewing/QA'ing #9666 I noticed that there is a mismatch in the sign on the number:

image

the saving's amount is '-1.63' but in the tooltip it shows '+1.63'... this is 

Nope... nope... that's a tilde. That's **almost unrecognizable as a tilde**

**Comment by jakehaugen** [comment]
> in the screenshot, I was able to see it because it's super big compared to what I see on the extension popup on my laptop monitor. @jakehaugen thoughts? I don't think this blocks anything but it threw my old ass eyes for a loop 😆

Yes, this is an issue with the font we are using. In the mocks I use a different font for the tilde -- Euclid Flex vs. Euclid Circular B. Huge pain. But, I thought we had it solved elsewhere?

**Comment by brad-decker** [comment]
@jakehaugen possibly, i just ran into it when testing. @danjm might have more insight.

**Comment by danjm** [comment]
The updates to the tooltip designs most recently been requested by @jakehaugen have been added to this PR:
Screenshot from 2020-10-26 11-14-06

**Comment by danjm** [comment]
The tilde has been fixed
Screenshot from 2020-10-26 11-47-54

**Comment by danjm** [comment]
This was updated to include desired messaging from @jakehaugen when there are no savings:

Screenshot from 2020-10-26 15-00-38

**Review by brad-decker — DISMISSED** [review]
LGTM

**Comment by Gudahtt** [comment]
Moving this into draft until #9611 is merged

**Comment by danjm** [comment]
@brad-decker This branch has now been updated with a rebase onto latest develop. With that change, it unifies the two halfs of the view quote screen redesign.

Two other notable changes that have also be made:
- The counter timer redesign is now included in this PR
- the savings UI added with this PR (the text and piggy bank in the header of the fee card box) is now hidden when the environment is not 'development'. This is because we still want to gather some analytics before we release that UI. However, it will be helpful to have it available on develop between now and when we decide to release, for UX testing purposes.

**Comment by danjm** [comment]
<details><summary>Screenshots of what this looks like now:</summary>







</details>

**Comment by darkwing** [comment]
The only big issue is seeing the "Terms of Service" link cut off on desktop, which I think is something we should fix -- we shouldn't risk users missing it:

TermsBlocked

At present, `.swaps__container` is `height: 620px;`; could we change that to `min-height` when in full page mode?

I'm a big concerned about the way we're using space here:

Box

Maybe we could use a sideways arrow (" -> ") to better use the space?

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (darkwing) [inline ui/app/pages/swaps/fee-card/fee-card.js:48]: If we initialize `savingsText` to `''`, we can void the last `else` and possibly the `if (!inDevelopment)` block.

File: ui/app/pages/swaps/fee-card/index.scss
Comment (darkwing) [inline ui/app/pages/swaps/fee-card/index.scss:161]: Might be helpful to add a comment here as to why we're using `!important`, not sure if we have a standard for doing that with `important` usage.

**Comment by jakehaugen** [comment]
> The only big issue is seeing the "Terms of Service" link cut off on desktop, which I think is something we should fix -- we shouldn't risk users missing it:
> 
> TermsBlocked
> 
> At present, `.swaps__container` is `height: 620px;`; could we change that to `min-height` when in full page mode?
> 
> I'm a big concerned about the way we're using space here:
> 
> Box
> 
> Maybe we could use a sideways arrow (" -> ") to better use the space?

Will soon be proposing that the Terms of Service link be moved to the first page. There are a few other tweaks in the works along with this. May require a different PR?

**Review by darkwing — CHANGES_REQUESTED** [review]
Added a few suggestions here.

File: ui/app/pages/swaps/fee-card/index.scss
Comment (danjm) [reply ui/app/pages/swaps/fee-card/index.scss:161]: Done in 5d6c28567

**Comment by danjm** [comment]
@darkwing The terms of service issue was addressed in 7ac38c3b5

Now looks like:

Screenshot from 2020-11-09 15-57-59

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (danjm) [reply ui/app/pages/swaps/fee-card/fee-card.js]: Done in 0c0dffdef

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (darkwing) [inline ui/app/pages/swaps/fee-card/fee-card.js:58]: `inDevelopment` and `tokenConversionRate` are checked in both else-ifs, so I think that could be tightened up, but I don't want to block on it.

**Review by darkwing — DISMISSED** [review]

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (brad-decker) [inline ui/app/pages/swaps/fee-card/fee-card.js:173]: could we make use of the wrapperClassName/containerClassName prop on the InfoTooltip component instead of manually creating the extra div here?

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (danjm) [reply ui/app/pages/swaps/fee-card/fee-card.js:173]: some day I will do this without you having to ask me about it on a PR :sweat_smile:

Comment (danjm) [reply ui/app/pages/swaps/fee-card/fee-card.js]: Done in a7d49f261

File: ui/app/pages/swaps/fee-card/fee-card.js
Comment (danjm) [reply ui/app/pages/swaps/fee-card/fee-card.js:58]: I'm going to leave this where it is for now, as we will be removing this variable in the not to distant future

**Review by brad-decker — APPROVED** [review]

**Pull request merged**
